scripts/gen-hie.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+~/.cabal/bin/gen-hie > hie.yaml
```

Comment:
We can't require `gen-hie` to be installed by `cabal`, so we need to require `gen-hie` be on the PATH. 

That means we need it to be available in our nix-shell as well, which I think is as easy as adding it to tooling: https://github.com/input-output-hk/cardano-ledger/blob/b9aa1ad1728c0ceeca62657ec94d6d099896c052/shell.nix#L33-L38

Reply:
Additing it to the `tools` didn't work (with `key not found` error), but adding it to the `nativeBuildInputs` worked :+1: